test/message.test.js

```diff
@@ -3,23 +3,67 @@ import { fixtureSync } from '@open-wc/testing-helpers';
 import '../vaadin-message.js';
 
 describe('message', () => {
-  let message, value;
+  let message, avatar, content, time, name, user;
 
   beforeEach(() => {
-    message = fixtureSync('<vaadin-message></vaadin-message>');
-    value = message.shadowRoot.querySelector('[part="value"]');
+    let root = document.documentElement;
+    root.style.setProperty('--vaadin-user-color-2', 'blue');
+
+    message = fixtureSync('<vaadin-message>Hello</vaadin-message>');
+    avatar = message.shadowRoot.querySelector('vaadin-avatar');
+    name = message.shadowRoot.querySelector('[part="name"]');
+    content = message.shadowRoot.querySelector('[part="content"]');
+    time = message.shadowRoot.querySelector('[part="time"]');
+    user = {
+      name: 'Joan Doe',
+      abbr: 'JD',
+      img: 'https://i.pravatar.cc/150?img=39',
+      colorIndex: 2
+    };
   });
 
   it('should have a valid version number', () => {
     expect(message.constructor.version).to.match(/^(\d+\.)?(\d+\.)?(\d+)(-(alpha|beta)\d+)?$/);
   });
 
-  it('should say initially hello world', () => {
-    expect(value.textContent).to.be.equal('hello world');
+  it('avatar should be initially visible but without data', () => {
+    expect(avatar.getAttribute('name')).to.be.null;
+    expect(avatar.getAttribute('abbr')).to.be.null;
+    expect(avatar.getAttribute('img')).to.be.null;
+    expect(avatar.getAttribute('title')).to.be.equal('anonymous');
+    expect(avatar.getAttribute('has-color-index')).to.be.null;
+  });
+
+  it('name should be initially empty', () => {
+    expect(name.textContent.trim()).to.be.equal('');
+  });
+
+  it('time should be initially empty', () => {
+    expect(time.textContent.trim()).to.be.equal('');
+  });
+
+  it('avatar should be set with provided user', () => {
+    message.user = user;
+    expect(avatar.getAttribute('name')).to.be.equal('Joan Doe');
+    expect(avatar.getAttribute('abbr')).to.be.equal('JD');
+    expect(avatar.getAttribute('img')).to.be.equal('https://i.pravatar.cc/150?img=39');
+    expect(avatar.getAttribute('title')).to.be.equal('Joan Doe (JD)');
+    expect(avatar.getAttribute('has-color-index')).to.be.not.null;
```

Comment:
Why `has-color-index` instead of the actual property that is set?

Reply:
when you set the color index, the avatar component does two things.
1. It sets the `has-color-index` **attribute**
2. It sets the inline **style**, like `--vaadin-avatar-user-color: var(--vaadin-user-color-3);`

Testing an attribute was simpler than testing an inline style, so I went with that. I can do both as well, it is not much extra.

Reply:
Ok, so `colorIndex` itself doesn't reflect to attribute. But it's still a property: `avatar.colorIndex`.

Reply:
Oh I did not know that as I was only inspecting DOM! Now I added a test to see that the number matches as well.